provisioning/baremetal_secrets_test.go

```diff
@@ -0,0 +1,87 @@
+package provisioning
+
+import (
+	"testing"
+
+	"golang.org/x/net/context"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	fakekube "k8s.io/client-go/kubernetes/fake"
+)
+
+const testNamespace = "test-namespce"
+
+func TestGenerateRandomPassword(t *testing.T) {
+	pwd, err := generateRandomPassword()
+	if err != nil {
+		t.Errorf("Unexpected error: %s", err)
+	}
+	if pwd == "" {
+		t.Errorf("Expected a valid string but got null")
+	}
+}
+
+//Testing the case where the password already exists
+func TestCreateMariadbPasswordSecret(t *testing.T) {
```

Comment:
Personally I am more inclined to simpler tests. This one also seems to combine together different cases, so I'd recommend splitting them also to increase readability.

Reply:
What this test is trying to test here is that fact that when mariadbPassword already exists, another call to this method would not create a new password. To test this behavior we need to make 2 calls to createMariadbPasswordSecret(). We are testing only one scenario here. 
I am open to splitting or making it simpler but not sure how. I can add comments within the code to explain what the test is trying to accomplish.

Reply:
Updated comment for the test to explain in detail what the test does.

Reply:
Let me try put down an example

Reply:
Here's the sample PR: https://github.com/sadasu/cluster-baremetal-operator/pull/2

Reply:
Thanks for the test you provided. I think this is a good addition to the existsing test so I have added it and kept the existing test that verifies functionality when we call the CreateMariadbPasswordSecret() multiple times when it already exists.

Reply:
Thanks Sandhya but honestly I don't see the point in keeping both the approaches (from a readability/maintenance point of viewe). In general, if there's a case missing it should be added in the list.

Reply:
Yes, I was able to add it as a case to this test.